components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -656,26 +647,40 @@ void JsonRpcService::UpdateIsEip1559(const std::string& chain_id,
   }
 }
 
-void JsonRpcService::FireNetworkChanged(mojom::CoinType coin) {
+void JsonRpcService::FireNetworkChanged(
+    mojom::CoinType coin,
+    const std::string& chain_id,
+    const absl::optional<url::Origin>& origin) {
   for (const auto& observer : observers_) {
-    observer->ChainChangedEvent(GetChainId(coin), coin);
+    observer->ChainChangedEvent(chain_id, coin, origin);
   }
 }
 
-std::string JsonRpcService::GetChainId(mojom::CoinType coin) const {
-  return chain_ids_.contains(coin) ? chain_ids_.at(coin) : std::string();
+std::string JsonRpcService::GetChainIdSync(
+    mojom::CoinType coin,
+    const absl::optional<::url::Origin>& origin) const {
+  return GetCurrentChainId(prefs_, coin, origin);
 }
 
-void JsonRpcService::GetChainId(
+void JsonRpcService::GetDefaultChainId(
     mojom::CoinType coin,
-    mojom::JsonRpcService::GetChainIdCallback callback) {
-  std::move(callback).Run(GetChainId(coin));
+    mojom::JsonRpcService::GetDefaultChainIdCallback callback) {
+  std::move(callback).Run(GetChainIdSync(coin, absl::nullopt));
+}
+
+void JsonRpcService::GetChainIdForOrigin(
+    mojom::CoinType coin,
+    const ::url::Origin& origin,
+    mojom::JsonRpcService::GetChainIdForOriginCallback callback) {
+  std::move(callback).Run(GetChainIdSync(coin, origin));
 }
 
 void JsonRpcService::GetBlockTrackerUrl(
     mojom::JsonRpcService::GetBlockTrackerUrlCallback callback) {
   std::move(callback).Run(
-      GetBlockTrackerUrlFromNetwork(GetChainId(mojom::CoinType::ETH)).spec());
+      GetBlockTrackerUrlFromNetwork(
+          GetChainIdSync(mojom::CoinType::ETH, absl::nullopt))
+          .spec());
 }
```

Comment:
I don't see any usage of this API, and this method is also the only caller of GetBlockTrackerUrlFromNetwork, can we remove these two?